### PR TITLE
Refactor AddressBook as InternshipData

### LINKED ISSUE #78: Refactor AddressBook as InternshipData
Refactor how persons were stored in the address book as internships stored in InternshipData.

### PR DESCRIPTION
**Pull request by theman-oj10**
Fixes #78, #25 
Implements the updated internship classes to work with the model package. Also updates the Sample Data to work with the new fields.

### CODE REVIEW

File: src/main/java/seedu/address/model/internship/exceptions/InternshipNotFoundException.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package seedu.address.model.internship.exceptions;
 
 /**
- * Signals that the operation is unable to find the specified internship.
+ * Signals that the operation is unable to find the specified person.
```
Comment (CaptainKeqing) [inline src/main/java/seedu/address/model/internship/exceptions/InternshipNotFoundException.java:4]: Should this be internship not person

File: src/main/java/seedu/address/model/internship/Location.java [hunk 2]
```diff
@@ -14,10 +14,11 @@ public class Location {
     /**
      * Enum of locations
      */
-    public enum LocationEnum {
+    private enum LocationEnum {
```
Comment (CaptainKeqing) [inline src/main/java/seedu/address/model/internship/Location.java:17]: should enums be public

File: src/main/java/seedu/address/model/internship/ApplicationStatus.java [hunk 3]
```diff
@@ -15,20 +15,21 @@ public class ApplicationStatus {
     /**
      * Enum of statuses
      */
-    public enum StatusEnum {
+    private enum StatusEnum {
```
Comment (CaptainKeqing) [inline src/main/java/seedu/address/model/internship/ApplicationStatus.java:18]: should this enum be public too

**Review by CaptainKeqing — COMMENTED** [review]
LGTM, just a few small nits

For future PR reference:
In general it took awhile to read through the whole change list because creating a new file and changing the fields from the old implementation does not highlight the diffs, so the reviewer has to look through the entire code instead of just the differences.

Also I think it will mess up the code line contribution tracker in the future but I guess it doesnt matter right now

File: src/main/java/seedu/address/model/internship/exceptions/InternshipNotFoundException.java (hunk 1 above)
Comment (theman-oj10) [reply src/main/java/seedu/address/model/internship/exceptions/InternshipNotFoundException.java]: Yes it has been fixed.

File: src/main/java/seedu/address/model/internship/Location.java (hunk 2 above)
Comment (theman-oj10) [reply src/main/java/seedu/address/model/internship/Location.java]: Yes it has been fixed

File: src/main/java/seedu/address/model/internship/ApplicationStatus.java (hunk 3 above)
Comment (theman-oj10) [reply src/main/java/seedu/address/model/internship/ApplicationStatus.java]: Yes it has been fixed.

File: src/main/java/seedu/address/model/internship/ApplicationStatus.java [hunk 4]
```diff
@@ -21,14 +21,15 @@ public enum StatusEnum {
         REJECTED,
         ACCEPTED,
         ONGOING
+
     }
 
-    public final StatusEnum applicationStatus;
+    private final StatusEnum applicationStatus;
```
Comment (CaptainKeqing) [inline src/main/java/seedu/address/model/internship/ApplicationStatus.java:27]: if private, then we need a getter for this

File: src/main/java/seedu/address/model/internship/Location.java [hunk 5]
```diff
@@ -17,10 +17,11 @@ public class Location {
     public enum LocationEnum {
         LOCAL,
         OVERSEAS,
-        REMOTE
+        REMOTE,
+        UNKNOWN
     }
 
-    public final LocationEnum location;
+    private final LocationEnum location;
```
Comment (CaptainKeqing) [inline src/main/java/seedu/address/model/internship/Location.java:24]: same comment as ApplicationStatus, we need a getter or public final

**Review by CaptainKeqing — CHANGES_REQUESTED** [review]

**Review by CaptainKeqing — APPROVED** [review]
LGTM, thanks for updating

**Pull request merged**